ecosystem/typescript/sdk/src/tests/e2e/fungible_asset_client.test.ts

```diff
@@ -0,0 +1,129 @@
13e5014e41c8084d0cb0a71de17350a9458ac66264b10d6e50500fe6db92882e155254dbd3a70e1303cb35fb4d22f1b78e40366808d49f163f749646244b894838d541b83d75773c707f7a4abb2f4d184751c94a124c074f3e957219cd661aa8399e8b1d48f8b39669f774cf523a5414c098c6059869d2cf04c180ce6b45ee9b7a8d2d5a677d19bd353f4fe5e679380d37d9aa7ead5763526220e9d41f1d97816707c7809fac82b69197040d2e4334bbaf1fd1fd597f6737bb19527159cc21d378c43fe877294940b15c0cb21ab16386cf2968b703ffb6a783c6847cf0b329f32aedcc54f69835b9adce8038cfd9ca8416e36ac154759d245cbd771c64d6990c20e957f50581c4a1dc55d3049e173c90d7696d1dab9a6603ab6e125dc7ae4bb781659d476d198d4d5e04e6759969989a376bd815ec5c956b17a193e1035c340e9697abc609a1a378d85af4ba0546b754dcfb704ec9aaad89ef88841ddae7384da1dc0a55ef74513407fb21204fb1fc5ed88c9ca2d529b1ada86d15e12925cb9c5d92d4c06659cd739ae862b320506baccec134b2cdd9b3bacb7bfe0d551cb5552b1cd76669dd6e5392759a7defb9f685db1a7b29da6fdf82899fb5d1a4157c3a54236a61ecf1c0cc0ad49a33701498fa1d83ea43d46ca19c6d9a3af0bcd73b382e24d031e9be50a3cf6b1760822858554addf2012a5954036a8a70f24f45e17104cc7da383b15dac7881e98f1ff5265cc953c6a1074ae33983967da54241d82a6958f7e011af2011eadd63250826a18af74ad005155204920df3cdf04012d2669b0c4bbfd346b4316487a1cf07028c09e8dbd48ca8f3d8184f373734bac2a0872dfb0e1e8e9be6773d2610f89be9403ad125248633552e6852eb09ca28c9d311ecd4543dfc649c150d681c0439d8faf270f24ac7e42628724cebb91ef4b81da81487af07571f5d8d660dbc5c1875f47330e5c0b6832b35b1f4ae373e2fefa8eab60937770a954ec3723398d41bbe0e447c57ead3156950cbe1b96fcc67af39213049c2908f93449120c110956aec2f591b3188a9763221f49c0ccfc83d0f669a716cb9f66b8859bfc3daf18583fd564eb4da6ef6a32ae3e0b78fd01064395e7801d9c422af48df957f94d9b8336fe1c9aafcdf9777e7cb0c66c20167be87949c72bcba931aadbcbab207acefee9c940b8dbcebc8ff687df61d5d9bf24675d42662dfdfbf984eefcedc6d3bc99eb79de62bc39e87bff948fe1a6d52af09ffa466fb95d64a3b95164525d57d96ba99219c9409a967707d5503cf27b5dd4e76da765d0ddde770e30beb44e8b95cefb1b714d420c6030a4cee1a43860d2d735aea7c3f3639b5e3a6dad502a45937c6e84df2860032c364102037858fe2cee577413dec52116b41cc641b5a5ddb219865817e3f858b37fdc9063eef105e5021d4609d9292c2acdd7e9e9876dc6bb55ee46c8ef36743723d6fc4f026ff89818a77cfa22e1782fa766d0f791702f6b3841e8880309e63411328086c15b8b7deccb7facd83af7bb76f79f997cc66b573c73160d1e1ad18733069bbf1c5ee458613d2b984d4d3aedd1ca207bbbba17fded07dcb9a4451a3612ca2c171188227f04ea94319eded19291e76a3daed611a71c725dd240515b6bc8fc911a2471d76cd383c7674e89a5a59f2336ce2f1e85a24448fbb77c7bdbeee136cf2dd2d8cf2633718d59c376e909f76dd84ba29d8ac095b94dae8bbf7815c2c2192d5c509a639948300431e90f089292540bd7bf1687777f741887296607dc3b887d467653f78fb3966430c6b9b29919af56f25b4cdcd9e126d6dc85e6f5d6ffd0b207ad810cb20000000000400000000000000000000000000000000000000000000000000000000000000010e4170746f734672616d65776f726b00000000000000000000000000000000000000000000000000000000000000010b4170746f735374646c696200000000000000000000000000000000000000000000000000000000000000010a4d6f76655374646c69620000000000000000000000000000000000000000000000000000000000000004114170746f73546f6b656e4f626a6563747300",
```

Comment:
May be not in the scope of this PR: This is hardcoded so if the contract ever changes, these tests would be outdated right? Have you explored any alternatives to doing these e2e tests?

Reply:
yes this is where it becomes tricky. we need to somehow compile it and then publish it so we can use this module in tests. to publish we can use the SDK function, but we most pre-compile it somehow before
1. compile locally and using this string
2. compile locally and include the compiled file and then read from it

both would have a static compiled result

Reply:
One idea I can come up with is change the test process to have a step that compiles and deploys the move-examples code against a local network via bash script (to invoke the CLI)

Reply:
Yeah